cmd/jaeger/internal/all-in-one.yaml

```diff
@@ -5,7 +5,19 @@ service:
       receivers: [otlp, jaeger, zipkin]
       processors: [batch]
       exporters: [jaeger_storage_exporter]
-
+  telemetry:
+    metrics:
+      level: detailed
+    traces:
+      processors:
+        - batch:
+            schedule_delay: 1000
+            exporter:
+              otlp:
+                endpoint: http://localhost:4318/v1/traces
+                protocol: http/protobuf
+                headers:
+                  Authorization: "Basic ..."
```

Comment:
Made some changes, without the rest of the configuration on running the binary it would show an error 
`Tracer Provider cannot be initialized`

Reply:
I had to revert back the changes, without the above configuration the test is failing.
Reference: [PR](https://github.com/open-telemetry/opentelemetry-collector/pull/10490)

Reply:
which part can you not remove? For example. why do we need Authorization (with seemingly invalid value)?

Reply:
On reverting the changes, the test is still failing. Debugging as to why it happened because in the last three commits the test didn't fail once and now suddenly after the latest push the getAPIV3 test is failing

Reply:
@yurishkuro the getAPIV3Trace is failing with the Error:  "git-commit=df61ec6b21b492b31f64e667c0a0e4f8a9cdb5f7, git-version=v1.59.0, build-date=2024-07-17T16:05:01Z" does not contain "jaeger"

Reply:
@yurishkuro the getAPIV3Trace is failing with the `Error:  "git-commit=df61ec6b21b492b31f64e667c0a0e4f8a9cdb5f7, git-version=v1.59.0, build-date=2024-07-17T16:05:01Z" does not contain "jaeger"`
Does it have to do something with the way tracer is configured?

Reply:
@yurishkuro the getAPIV3Trace is failing with the `Error:  "git-commit=df61ec6b21b492b31f64e667c0a0e4f8a9cdb5f7, git-version=v1.59.0, build-date=2024-07-17T16:05:01Z" does not contain "jaeger"`
Does it have to do something with the way tracer is configured?

Reply:
On manually creating a trace, the /metrics endpoint is showing the service name as Jaeger evident below in the screenshot.
![Screenshot 2024-07-17 223240](https://github.com/user-attachments/assets/105f5729-1617-43f7-8fb0-cccfdcf7806b)

Reply:
![Screenshot 2024-07-17 231106](https://github.com/user-attachments/assets/88895c19-6799-4dd1-bc8b-40cbf01c5eed)
These are the logs of the metric endpoint by just setting level as detailed, I think parity is obtained in metrics in internal observability already.

Reply:
the correct response from all-in-one v1 is
```
Response body: {"services":["jaeger-all-in-one"]}
```

from v2:
```
Response body: {"services":["git-commit=, git-version=, build-date="]}
```
somehow we are passing a weird string as the service name when creating the tracer provider.

> I think parity is obtained in metrics in internal observability already.

Reply:
Yes exactly

Reply:
@yurishkuro I think I have figured it out. That string is being initialized from the OTEL Tracer Provider code itself.
[Reference](https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/telemetry/tracer.go)
![Screenshot 2024-07-18 002348](https://github.com/user-attachments/assets/2a119835-a0d3-4fae-b0cc-0cf895c96586)

Reply:
looks like a bug in OTEL collector

Reply:
Yeah guess so. What should I do next?

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/10644

Reply:
You can temporarily fix it by setting Version to `jaeger` in command.go

Reply:
I pushed a fix

Reply:
Pardon me, I pushed the fix at the same time resulting in conflict